src/ophyd_async/sim/demo/sim_motor.py

```diff
@@ -0,0 +1,122 @@
+import asyncio
+import time
+from typing import Callable, List, Optional
+
+from bluesky.protocols import Movable, Stoppable
+
+from ophyd_async.core import StandardReadable
+from ophyd_async.core.async_status import AsyncStatus
+from ophyd_async.core.signal import soft_signal_r_and_backend, soft_signal_rw
+
+
+class SimMotor(StandardReadable, Movable, Stoppable):
+    def __init__(self, prefix: str, name="", instant=True) -> None:
+        """
+        Simulated motor device
+
+        args:
+        - prefix: str: Signal names prefix
+        - name: str: name of device
+        - instant: bool: whether to move instantly, or with a delay
+        """
+        self._instant = instant
+        self._move_task: Optional[asyncio.Task] = None
+
+        # Define some signals
+        self.user_setpoint = soft_signal_rw(
+            float, "user_setpoint", prefix + ".setpoint", initial_value=0
+        )
```

Comment:
After @jsouter makes the changes then this will be:
```suggestion
        self.user_setpoint = soft_signal_rw(float, initial_value=0)
```

Reply:
OK -  i already did this and checked it works against the latest of James branch - just not rebased this PR yet.

However - why is there no name at all? I also added the prefix to the name - if this is not needed should I drop prefix from the constructor?

Reply:
James has now removed the required argument 'name' and I have dropped prefix from the SimMotor constructor.